hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1907,20 +1908,54 @@ func (r *NodePoolReconciler) reconcileMachineHealthCheck(ctx context.Context,
 		if _, err := intstr.GetScaledValueFromIntOrPercent(&maxUnhealthyValue, 100, true); err != nil {
 			log.Error(err, "Cannot parse max unhealthy override duration", "value", maxUnhealthyOverride)
 		} else {
-			maxUnhealthy = maxUnhealthyValue
+			maxUnhealthy = &maxUnhealthyValue
 		}
 	}
 
-	timeOutOverride := nodePool.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation]
```

Comment:
We want to keep these as annotations? Or add a field for this as well?

Reply:
yes we are keeping them for now: https://github.com/openshift/hypershift/pull/4160#discussion_r1627667189
The timeout field is defined per Unhealthy Condition

Reply:
Er - right now you're `unhealthyConditions = append(unhealthyConditions` - are we requiring that the user remove the annotation *and* update the spec in the same UPDATE call? Perhaps logic to make sure that we don't end up with duplicates based on the current append logic is good.

Reply:
the annotation will be ignored if the unhealthyConditions is specified.
I am resetting the array before the append `unhealthyConditions = []capiv1.UnhealthyCondition{}`